source/code/plugins/in_npmd_server.rb

```diff
@@ -242,19 +249,47 @@ def npmd_reader
             end while !@npmdClientSock.nil?
         end
 
-        def package_and_send_error_log(msg)
-            _dataItems = Array.new
+        def is_valid_dataitem(item)
+            _itemType=""
+            if item["SubType"] == "NetworkAgent"
+                _itemType = NPMContract::DATAITEM_AGENT
+            elsif item["SubType"] == "NetworkPath"
+                _itemType = NPMContract::DATAITEM_PATH
+            elsif item["SubType"] == NPM_DIAG
+                _itemType = NPMContract::DATAITEM_DIAG
+            end
+
+            return false if _itemType.empty?
+
+            _res, _prob = NPMContract::IsValidDataitem(item, _itemType)
+
+            return true if _res == NPMContract::DATAITEM_VALID
+            if (_res == NPMContract::DATAITEM_ERR_INVALID_FIELDS)
+                Logger::logInfo "Invalid key in #{item["SubType"]} data: #{_prob}"
+            elsif (_res == NPMContract::DATAITEM_ERR_MISSING_FIELDS)
+                Logger::logInfo "Key #{_prob} absent in #{item["SubType"]} data"
+            end
```

Comment:
nitpick: although you are checking for valid itemtype [lines 253-261], it would be a good practice to add a log in case _res is DATAITEM_ERR_INVALID_TYPE because it is a possible return value for NPMContract::IsValidDataitem